### PR TITLE
Change all herring to smddp

### PR DESCRIPTION
**Pull request by rondogency**
*Description of changes:*
Change herringcommon to smddpcommon for API fix
Change other herring codename

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: models/nlp/albert/requirements.sh [hunk 1]
```diff
@@ -6,5 +6,5 @@ pip install --no-cache-dir \
         pandas \
         apache_beam \
         pyarrow==0.16 \
-        git+https://github.com/HerringForks/transformers.git@master \
+        git+https://github.com/smddpForks/transformers.git@master \
```
Comment (ChaiBapchya) [inline models/nlp/albert/requirements.sh]: ?

File: models/nlp/albert/run_pretraining.py [hunk 2]
```diff
@@ -64,11 +64,13 @@
 )
 
 # See https://github.com/huggingface/transformers/issues/3782; this import must come last
-import smdistributed.dataparallel.tensorflow as herring  # isort:skip
-import herringcommon as hc
+import smdistributed.dataparallel.tensorflow as smddp  # isort:skip
+
+import smddpcommon
```
Comment (ChaiBapchya) [inline models/nlp/albert/run_pretraining.py]: shouldn't we refactor to not use `smddpcommon`? We don't expose that API to public.

**Review by ChaiBapchya — CHANGES_REQUESTED** [review]
incorrect repo name change

File: models/nlp/albert/run_pretraining.py (hunk 2 above)
Comment (rondogency) [reply models/nlp/albert/run_pretraining.py]: no you have refactored already and there is not herringcommon api in our release, but this thing is needed for setting up the bucket size

File: models/nlp/albert/requirements.sh (hunk 1 above)
Comment (rondogency) [reply models/nlp/albert/requirements.sh]: good catch, thanks

File: models/nlp/albert/run_pretraining.py (hunk 2 above)
Comment (ChaiBapchya) [reply models/nlp/albert/run_pretraining.py]: Yes ofcourse herringcommon is not to be used. I refactored it :)
Yes I know it is being used to set bucket size. 

Rephrasing my above question; shouldn't we "rewrite" above logic to NOT use smddpcommon. Speak: rewrite the logic that sets bucket size using gradient tape rather than use smddpcommon. Because we dont expose smddpcommon API to public

File: models/nlp/albert/run_pretraining.py [hunk 3]
```diff
@@ -66,9 +66,13 @@
 # See https://github.com/huggingface/transformers/issues/3782; this import must come last
 import smdistributed.dataparallel.tensorflow as smddp  # isort:skip
 
-import smddpcommon
+# TODO : Change to obfuscate smddpcommon. This code does not use GradientTape, so need to pass it like this.
 bucket_cap_bytes = int(64 * 1024 * 1024)
-smddpcommon.setBucketSize(bucket_cap_bytes) # TODO : Change to obfuscate smddpcommon. This code does not use GradientTape, so need to pass it like this.
```
Comment (ChaiBapchya) [inline models/nlp/albert/run_pretraining.py]: we are still using smddpcommon/herringcommon and thus the comment still remains valid. why are we removing it?

**Comment by ChaiBapchya** [comment]
`bucket_cap_mb` parameter passed to Distributed Optimizer class should be the right way of setting bucket size instead of using an internal shared library [lib<abc>_common] which is not exposed to public

**Review by ChaiBapchya — APPROVED** [review]
Currently, this script invokes smddp allreduce directly instead of via smddp optimizer/ Gradient tape.
It's a non-trivial change to update this training script [since it will diverge from the upstream training script]. Approving this PR since it serves its purpose for now. Requires further discussion to finalize whether to continue with this approach or diverge from upstream
